src/refs/immutable.rs

```diff
@@ -19,7 +20,7 @@ use crate::borrow::AtomicBorrow;
 /// [`AtomicCell`]: struct.AtomicCell.html
 /// [`&T`]: https://doc.rust-lang.org/core/primitive.reference.html
 pub struct Ref<'a, T: ?Sized> {
-    value: &'a T,
+    value: NonNull<T>,
```

Comment:
`NonNull` is not `Send`/`Sync` so you will probably want to manually implement this on `Ref`/`RefMut` (assuming that `AtomicBorrow` is `Send`/`Sync`), see my PR on `atomic_refcell` for reference.